### PR TITLE
gdrive 2.1.0 (migrate from boneyard)

### PR DESCRIPTION
**Pull request by paulz**
- [X] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally prior to submission with `brew install gdrive`?
- [X] Does your submission pass `brew audit --strict --online gdrive` (after doing `brew install gdrive`)?
## Google Drive Command Line Interface

Updated formula to fix:
https://github.com/Homebrew/legacy-homebrew/issues/45916

### CODE REVIEW

**Comment by paulz** [comment]
Request to remove *gdrive* from boneyard: https://github.com/Homebrew/homebrew-boneyard/pull/111

**Comment by ilovezfs** [comment]
How have the original reason for boneyarding changed?

**Comment by paulz** [comment]
Original reason for removing were no maintainers and formulas fails to build under the latest tools.
We now have up-to-date formula that I am willing to maintain.

**Comment by ilovezfs** [comment]
Thanks for the explanation.

It appears that upstream vendors its own resources. Have you attempted to get this working without resorting to `go_resource`?

**Comment by paulz** [comment]
@ilovezfs yes, and I've been maintaining [this formula on tap](https://github.com/paulz/homebrew-gdrive/blob/master/gdrive.rb):

You can try out latest formula on tap:
```
brew tap paulz/drive
brew install drive
```

[Another commit removing previous gdrive](https://github.com/Homebrew/legacy-homebrew/pull/46301/commits/ae1f44244bddeac8d02df0da07fbc168a1bd6c3f)

**Comment by ilovezfs** [comment]
And what was the problem with not using `go_resource` at all?

File: Formula/gdrive.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  end
+
+  depends_on "go" => :build
+  depends_on "godep" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+    mkdir_p buildpath/"src/github.com/prasmussen/"
+    ln_sf buildpath, buildpath/"src/github.com/prasmussen/gdrive"
+    Language::Go.stage_deps resources, buildpath/"src"
```
Comment (ilovezfs) [inline Formula/gdrive.rb]: This line can be removed.

File: Formula/gdrive.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
+require "language/go"
+
```
Comment (ilovezfs) [inline Formula/gdrive.rb]: These two lines can be removed.

**Comment by ilovezfs** [comment]
I tested your formula locally with my Google drive, and it does indeed work. Cool.

File: Formula/gdrive.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    mkdir_p buildpath/"src/github.com/prasmussen/"
+    ln_sf buildpath, buildpath/"src/github.com/prasmussen/gdrive"
+    Language::Go.stage_deps resources, buildpath/"src"
+    system "godep", "go", "build", "-o", "gdrive", "."
+    bin.install "gdrive"
+    doc.install "README.md"
+  end
+
+  test do
+    system "#{bin}/gdrive", "version"
```
Comment (ilovezfs) [inline Formula/gdrive.rb]: Do this instead please:
```
assert_match version.to_s, shell_output("#{bin}/gdrive version")
```

File: Formula/gdrive.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
+class Gdrive < Formula
+  desc "Google Drive CLI Client"
+  homepage "https://github.com/prasmussen/gdrive"
+  url "https://github.com/prasmussen/gdrive/archive/2.1.0.tar.gz"
+  sha256 "a1ea624e913e258596ea6340c8818a90c21962b0a75cf005e49a0f72f2077b2e"
+  head "https://github.com/prasmussen/gdrive.git"
+
+  bottle do
```
Comment (DomT4) [inline Formula/gdrive.rb]: Don't need to add this block. Will get done automagically on pull.

Comment (paulz) [inline Formula/gdrive.rb]: done

File: Formula/gdrive.rb [hunk 5]
```diff
@@ -0,0 +1,22 @@
+class Gdrive < Formula
```
Comment (paulz) [inline Formula/gdrive.rb]: This file should be the same as one we use on tap:
 https://github.com/paulz/homebrew-gdrive/blob/master/gdrive.rb

See https://github.com/paulz/homebrew-gdrive#usage
```
brew tap paulz/gdrive
brew install gdrive
```

**Comment by ilovezfs** [comment]
:+1:

**Comment by DomT4** [comment]
Merged in https://github.com/Homebrew/homebrew-core/commit/5e1e1638346bc08cba75eaa54f7a75ee04fe8287. Thank you for your contribution to Homebrew @paulz; we appreciate it! :smiley_cat:

**Comment by paulz** [comment]
Thank you Homebrew team!

Happy to maintain the formula, will send you pull requests. 😊

**Pull request closed without merging**